WalletWasabi.Fluent/ViewModels/Wallets/MusicBoxCoinjoinProfileSelectorViewModel.cs

```diff
@@ -0,0 +1,96 @@
+using ReactiveUI;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Fluent.Helpers;
+using WalletWasabi.Fluent.ViewModels.CoinJoinProfiles;
+using WalletWasabi.Fluent.ViewModels.Dialogs;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets;
+
+[NavigationMetaData(Title = "Coinjoin Profiles")]
+public partial class MusicBoxCoinjoinProfileSelectorViewModel : RoutableViewModel
+{
+	[AutoNotify] private CoinJoinProfileViewModelBase? _selectedProfile;
+	[AutoNotify] private bool _isFlyoutOpen;
+	[AutoNotify] private bool _showProfile;
+
+	public MusicBoxCoinjoinProfileSelectorViewModel(KeyManager keyManager)
+	{
+		Profiles = DefaultProfiles.ToList();
+		Profiles.Add(new ManualCoinJoinProfileViewModel(keyManager));
+
+		_selectedProfile = IdentifySelectedProfile(keyManager);
+
+		OpenFlyoutCommand = ReactiveCommand.Create(() =>
+		{
+			IsFlyoutOpen = false;
+			IsFlyoutOpen = true;
+		});
+		SelectProfileCommand = ReactiveCommand.CreateFromTask<CoinJoinProfileViewModelBase>(async p => await OnCoinjoinProfileSelectedAsync(keyManager, p));
+
+		this.WhenAnyValue(x => x.SelectedProfile)
+			.Select(x => x is null)
+			.CombineLatest(Services.UiConfig.WhenAnyValue(x => x.PrivacyMode))
```

Comment:
This is a potential memory leak. Please dispose the subscription once it's no longer needed.-

Reply:
@SuperJMN fixed in latest commit